integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py

```diff
@@ -428,8 +428,9 @@ def _query_result_to_documents(result: QueryResult) -> List[List[Document]]:
                 }
 
                 # prepare metadata
-                if metadatas := result.get("metadatas"):
-                    document_dict["meta"] = dict(metadatas[i][j])
+                metadatas = result.get("metadatas")
+                if metadatas and metadatas[i][j] is not None:
```

Comment:
Can it ever happen that `metadatas` size is shorter than `i`? And `metadatas[i]` is shorter than `j`? 🤔 

Cause if it can happen this could fail.

Reply:
Added an exception here as discussed.